message_ix_models/tools/iea/weo.py

```diff
@@ -0,0 +1,672 @@
+            & (df_costs.cost_type == "annual_om_costs"),
+            "cost_NAM_adjusted",
+        ] = dict_fom[f]
+
+
+# Type 4: function of another cost value (using ratio)
+def calc_nam_cost_ratio(
+    df_costs, desired_tech, desired_cost_type, reference_tech, reference_cost_type
+):
+    c_adj_ref = df_costs.loc[
+        (df_costs.message_technology == reference_tech)
+        & (df_costs.cost_type == reference_cost_type),
+        "cost_NAM_adjusted",
+    ].values[0]
+
+    orig_des = df_costs.loc[
+        (df_costs.message_technology == desired_tech)
+        & (df_costs.cost_type == desired_cost_type),
+        "cost_NAM_original_message",
+    ].values[0]
+
+    orig_ref = df_costs.loc[
+        (df_costs.message_technology == reference_tech)
+        & (df_costs.cost_type == reference_cost_type),
+        "cost_NAM_original_message",
+    ].values[0]
+
+    c_adj_des = c_adj_ref * (orig_des / orig_ref)
+
+    df_costs.loc[
+        (df_costs.message_technology == desired_tech)
+        & (df_costs.cost_type == desired_cost_type),
+        "cost_NAM_adjusted",
+    ] = c_adj_des
+
+    # return c_adj_des
+
+
+dict_tech_ref_inv = {
+    "gas_ppl": {"reference_tech": "gas_cc", "reference_cost_type": "capital_costs"},
+    "meth_coal": {"reference_tech": "syn_liq", "reference_cost_type": "capital_costs"},
+    "syn_liq_ccs": {
+        "reference_tech": "syn_liq",
+        "reference_cost_type": "capital_costs",
+    },
+    "meth_coal_ccs": {
+        "reference_tech": "meth_coal",
+        "reference_cost_type": "capital_costs",
+    },
+    "h2_bio": {"reference_tech": "h2_coal", "reference_cost_type": "capital_costs"},
+    "h2_bio_ccs": {"reference_tech": "h2_bio", "reference_cost_type": "capital_costs"},
+    "eth_bio": {"reference_tech": "liq_bio", "reference_cost_type": "capital_costs"},
+    "eth_bio_ccs": {
+        "reference_tech": "eth_bio",
+        "reference_cost_type": "capital_costs",
+    },
+    "solar_th_ppl": {
+        "reference_tech": "solar_pv_ppl",
+        "reference_cost_type": "capital_costs",
+    },
+    "solar_pv_I": {
+        "reference_tech": "solar_pv_ppl",
+        "reference_cost_type": "capital_costs",
+    },
+    "solar_pv_RC": {
+        "reference_tech": "solar_pv_ppl",
+        "reference_cost_type": "capital_costs",
+    },
+    "meth_ng": {"reference_tech": "syn_liq", "reference_cost_type": "capital_costs"},
+    "meth_ng_ccs": {
+        "reference_tech": "meth_ng",
+        "reference_cost_type": "capital_costs",
+    },
+    "coal_ppl_u": {
+        "reference_tech": "coal_ppl",
+        "reference_cost_type": "capital_costs",
+    },
+    "liq_bio_ccs": {
+        "reference_tech": "liq_bio",
+        "reference_cost_type": "capital_costs",
+    },
+    "solar_i": {
+        "reference_tech": "solar_pv_ppl",
+        "reference_cost_type": "capital_costs",
+    },
+}
+
+dict_tech_ref_fom = {
+    "gas_ppl": {"reference_tech": "gas_cc", "reference_cost_type": "annual_om_costs"},
```

Comment:
Looks like "reference_" in the string keys could safely be dropped.
Use a comment where these are used to indicate these are references.